InnerEye/ML/deep_learning_config.py

```diff
@@ -608,6 +611,40 @@ def get_path_to_checkpoint(self, epoch: int, for_mean_teacher_model: bool = Fals
                / self.checkpoint_folder \
                / f"{epoch}{filename}"
 
+    def get_recovery_path_train(self, run_recovery: Optional[RunRecovery],
+                                is_mean_teacher: bool, epoch: int) -> Path:
+        if run_recovery:
+            checkpoint_paths = run_recovery.get_checkpoint_paths(epoch, is_mean_teacher)[0]
+        else:
+            checkpoint_paths = self.get_path_to_checkpoint(epoch, is_mean_teacher)
+        return checkpoint_paths
+
+    def get_recovery_path_test(self, run_recovery: Optional[RunRecovery],
+                               is_mean_teacher: bool, epoch: int) -> List[Path]:
+        if run_recovery:
+            checkpoint_paths = run_recovery.get_checkpoint_paths(epoch, is_mean_teacher)
+            checkpoint_exists = []
+            # Discard any checkpoint paths that do not exist - they will make inference/registration fail.
+            # This can happen when some child runs fail; it may still be worth running inference
+            # or registering the model.
+            for path in checkpoint_paths:
+                if path.is_file():
+                    checkpoint_exists.append(path)
+                else:
+                    logging.warning(f"Could not recover checkpoint path {path}")
+
+            if len(checkpoint_exists) > 0:
+                return checkpoint_paths
```

Comment:
If we had a run that was recovered from another and continues training, it will reach the inference stage with a valid run recovery object, but no checkpoint for the last epoch. We can refactor further so that we never see this kind of situation, but I think its beyond the scope of this PR.